firmware/application/transmitter_model.hpp

```diff
@@ -26,77 +27,79 @@
 #include <cstdint>
 #include <cstddef>
 
-#include "receiver_model.hpp"
 #include "app_settings.hpp"
+#include "max2837.hpp"
 #include "message.hpp"
+#include "receiver_model.hpp"
 #include "rf_path.hpp"
-#include "max2837.hpp"
-#include "volume.hpp"
 #include "signal.hpp"
 
 class TransmitterModel {
    public:
+    static constexpr uint32_t default_baseband_bandwidth = max283x::filter::bandwidth_minimum;
+    static constexpr uint32_t default_sampling_rate = 3'072'000;
+    static constexpr uint32_t default_channel_bandwidth = 1;
+    /* 35 should give approx 1m transmission range. */
+    static constexpr uint8_t default_tx_gain = 35;
+    static constexpr uint8_t default_gain = 0;
+    static constexpr bool default_amp = false;
+
     /* The frequency to transmit on. */
     rf::Frequency target_frequency() const;
     void set_target_frequency(rf::Frequency f);
 
-    void set_antenna_bias();
-
-    bool rf_amp() const;
-    void set_rf_amp(bool enabled);
-
-    // TODO: does this make sense on TX?
-    int32_t lna() const;
-    void set_lna(int32_t v_db);
-
     uint32_t baseband_bandwidth() const;
     void set_baseband_bandwidth(uint32_t v);
 
-    // TODO: does this make sense on TX?
-    int32_t vga() const;
-    void set_vga(int32_t v_db);
-
-    int32_t tx_gain() const;
-    void set_tx_gain(int32_t v_db);
-
     // TODO: Doesn't actually affect radio.
     uint32_t channel_bandwidth() const;
     void set_channel_bandwidth(uint32_t v);
 
-    // TODO: does this make sense on TX?
     uint32_t sampling_rate() const;
     void set_sampling_rate(uint32_t v);
 
+    uint8_t tx_gain() const;
+    void set_tx_gain(uint8_t v_db);
+
+    // TODO: does this make sense on TX?
+    uint8_t lna() const;
+    void set_lna(uint8_t v_db);
+
+    // TODO: does this make sense on TX?
+    uint8_t vga() const;
+    void set_vga(uint8_t v_db);
+
+    bool rf_amp() const;
+    void set_rf_amp(bool enabled);
+
+    void set_antenna_bias();
+
     void enable();
     void disable();
 
-    /* Sets the model values without updating the radio. */
-    void set_configuration_without_update(
-        uint32_t baseband_bandwidth,
-        uint32_t sampling_rate);
+    void initialize();
 
     void configure_from_app_settings(const app_settings::AppSettings& settings);
 
    private:
-    bool enabled_{false};
-    bool rf_amp_{false};
-    int32_t lna_gain_db_{0};
-    uint32_t channel_bandwidth_{1};
-    uint32_t baseband_bandwidth_{max2837::filter::bandwidth_minimum};
-    int32_t vga_gain_db_{8};
```

Comment:
This is the one value that I didn't keep (it's 0 now). Is LNA/VGA even doing anything in the case of transmit anyway? I don't think they do.

Reply:
LNA is apparently applicable to transmit, whereas VGA is not, per HackRF docs.

Reply:
The RF_Amp is a shared concept for both. There’s a separate tx_gain that I think is the wrapper for LNA in TX mode (goes to 47). I think LNA and VGA are only RX.